File: howl-application/src/main/java/io/quartic/howl/GcsStorageBackendConfig.java
package io.quartic.howl;

import io.quartic.howl.storage.GcsStorageBackend;
import io.quartic.howl.storage.StorageBackend;

import java.io.IOException;
import java.security.GeneralSecurityException;

public class GcsStorageBackendConfig implements StorageBackendConfig {
    private String bucketName;

    public void setBucketName(String bucketName) {
        this.bucketName = bucketName;
    }

    @Override
    public StorageBackend build() {
        try {
            return new GcsStorageBackend(bucketName);
        } catch (IOException | GeneralSecurityException e) {
            throw new RuntimeException("can't construct google cloud storage backend: " + e);
        }
    }
}

File: howl-application/src/main/kotlin/io/quartic/howl/GcsStorageBackendConfig.kt
package io.quartic.howl

import io.quartic.howl.storage.GcsStorageBackend

data class GcsStorageBackendConfig(val bucketSuffix: String) : StorageBackendConfig {
    override fun build() = try {
        GcsStorageBackend(bucketSuffix)
    } catch (e: Exception) {
        throw RuntimeException("Can't construct Google Cloud Storage backend", e)
    }
}
